src/pow.cpp

```diff
@@ -161,21 +161,7 @@ bool CheckProofOfWork(uint256 hash, unsigned int nBits, const Consensus::Params
     return true;
 }
 
-arith_uint256 GetBlockProof(const CBlockIndex &block)
```

Comment:
Are the changes you made WRT this function due to linker issues when compiling cashlib? If not i dont understand the reason for this change

Reply:
yes.  I wanted to remove CBlockIndex from the android build, but still provide this functionality.  The only use of CBlockIndex in this function is to get the "nBits" field.  So, without changing semantics, I refactored this call into a new function GetWorkForDifficultyBits that just takes the nBits value.  cashlib pulls in GetWorkForDifficultyBits, but not this function.

  There are other dependencies I didn't want in pow.cpp, so I placed the new function into block.cpp, which is not a bad place for it because its calculating a block's work.